controllers/reconcilers/kserve_raw_metrics_service_reconciler.go

```diff
@@ -71,15 +74,58 @@ func (r *KserveMetricsServiceReconciler) Reconcile(ctx context.Context, log logr
 	return nil
 }
 
-func (r *KserveMetricsServiceReconciler) createDesiredResource(log logr.Logger, isvc *kservev1beta1.InferenceService) (*v1.Service, error) {
-	return nil, nil
+func (r *KserveRawMetricsServiceReconciler) createDesiredResource(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) (*v1.Service, error) {
+
+	isvcRuntime, err := utils.FindSupportingRuntimeForISvc(ctx, r.client, log, isvc)
+	if err != nil {
+		return nil, err
+	}
+
+	if isvcRuntime.Spec.Annotations == nil || isvcRuntime.Spec.Annotations[constants.PrometheusPortAnnotationKey] == "" {
+		log.V(1).Info("No Prometheus annotations on ServingRuntime, skipping creation of metrics resources")
+		return nil, nil
+	}
+
+	prometheusPortAnnotationValue := isvcRuntime.Spec.Annotations[constants.PrometheusPortAnnotationKey]
+	prometheusPort, err := strconv.ParseInt(prometheusPortAnnotationValue, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	metricsService := &v1.Service{
```

Comment:
Hmm this puts the onus of enabling metrics for a inferenceservice on the creator of the inferenceservice, rather than the platform. Even if we add logic to inject this in the UI flow, if a user is using yamls then they would have to specifically remember to enable metrics. In ODH/RHOAI, I'm guessing we always want metrics and the user should not have to worry about remembering to enable them. 
But I can be convinced otherwise :) 
cc @danielezonca @terrytangyuan WDYT

Reply:
Well, we could enable it by default in the ConfigMap: https://github.com/opendatahub-io/kserve/blob/ae30b971320f45bc98dd362b6b8873d7c1a48873/config/configmap/inferenceservice.yaml#L447. But doing that would also affect Serverless deployments, so not sure if it will impact something.

What I don't know is if this is specific to Serverless, because docs only refer to the Serverless use-case.

In case it works for Raw, I said _to inject_ meaning we could use a mutating webhook in odh-model-controller to inject the annotation on the ISVC. I wasn't meaning that the UI should do it.

Reply:
@israel-hdez revisiting this comment, we actively moved away from the kserve native annotations for metrics i.e `serving.kserve.io/enable-prometheus-scraping: "true"` for Serverless, and we have instead opted for `prometheus.io/port` and `prometheus.io/path` annotations on the ServingRuntime instead. 
So going back to `serving.kserve.io/enable-prometheus-scraping: "true"` for RawDeployment would IMO be more confusing. Creating a SeviceMonitor and Service for metrics will allow us to be consistent w.r.t metrics within RHOAI.